### PR TITLE
chore: migrate operation rules to rule authoring v2

### PR DESCRIPTION
**Pull request by niclim**
Migrating the [operation rules](https://github.com/snyk/sweater-comb/blob/main/src/rulesets/operation.ts) in this PR.

This should be the last one to migrate. There's a few things we should do before making the switch over to the new rule running API:
- I've got a change coming from optic to provide a way to write custom matcher error messages. Doing a `operationAssertions.requirement.matches({ summary: Matchers.string })` assertion will just show the operation and expected shape, but in this case, you can just add an error message that says operations must have a summary.
- Connect up the existing e2e tests to the new ruleset tests (there appears to be two, [internal JS tests](https://github.com/snyk/sweater-comb/blob/main/src/rulesets/tests/end-end.test.ts) and [full CLI tests](https://github.com/snyk/sweater-comb/tree/main/end-end-tests) - we'll want to run both, and spot check some of the error messages + runs to see if there's anything we want to address here

### CODE REVIEW

File: src/rulesets-new/operation-rules.ts [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+        const name = parameter.value.name;
+        if (name === "group_id" || name === "org_id") {
+          if (!parameter.value.schema) {
+            throw new RuleError({
+              message: "expected parameter to have a schema",
+            });
+          }
+
+          if (
+            !("$ref" in parameter.value.schema) &&
```
Comment (cmars) [inline src/rulesets-new/operation-rules.ts:186]: Please remind me, are refs guaranteed to be resolved? We may easily reuse org and group parameters with $refs.

**Review by cmars — APPROVED** [review]
LGTM. Really like the table tests!

File: src/rulesets-new/operation-rules.ts (hunk 1 above)
Comment (niclim) [reply src/rulesets-new/operation-rules.ts:186]: Yes, with a caveat with circular references (circular references will have a `$ref` value)

In the circular reference case, as long as we dereference at _least_ one level, rules will run against the dereferenced values.

Right now, optic-ci doesn't let you use circular references, in the next version release, we'll allow circular references with at least one level dereferenced, so you will always run rules on at least one level deep.

Do you know if there's any case you use (or will use) circular refs. I'm not sure if all the rules here have been written to handle these cases.

Comment (cmars) [reply src/rulesets-new/operation-rules.ts:186]: I think we've avoided circular references... that sounds wild. No, we haven't tried that. Not sure we would want to!

Now I'm curious, are there valid and useful applications for circular refs in OpenAPI? I don't believe I've ever seen this in the wild.

Comment (niclim) [reply src/rulesets-new/operation-rules.ts:186]: I think there are usecases for graph APIs. It's fairly uncommon to see it in regular REST APIs, but it does happen and we don't want optic-ci to crash when we detect one circular ref!

**Pull request merged**